app/controllers/api/users.php

```diff
@@ -1081,6 +1196,52 @@ function createUser(string $hash, mixed $hashOptions, string $userId, ?string $e
         $response->dynamic(new Document($prefs), Response::MODEL_PREFERENCES);
     });
 
+App::patch('/v1/users/:userId/targets/:targetId/identifier')
+    ->desc('Update user target\'s identifier')
+    ->groups(['api', 'users'])
+    ->label('audits.event', 'users.targets.update')
+    ->label('audits.resource', 'target/{response.$id}')
+    ->label('scope', 'targets.write')
+    ->label('sdk.auth', [APP_AUTH_TYPE_KEY, APP_AUTH_TYPE_ADMIN])
+    ->label('sdk.namespace', 'users')
+    ->label('sdk.method', 'updateTargetIdentifier')
+    ->label('sdk.description', '/docs/references/users/update-target-identifier.md')
+    ->label('sdk.response.code', Response::STATUS_CODE_OK)
+    ->label('sdk.response.type', Response::CONTENT_TYPE_JSON)
+    ->label('sdk.response.model', Response::MODEL_TARGET)
+    ->param('userId', '', new UID(), 'User ID.')
+    ->param('targetId', '', new UID(), 'Target ID.')
+    ->param('identifier', '', new Text(Database::LENGTH_KEY), 'The target identifier (token, email, phone etc.)')
+    ->inject('response')
+    ->inject('dbForProject')
+    ->inject('events')
+    ->action(function (string $userId, string $targetId, string $identifier, Response $response, Database $dbForProject, Event $events) {
+
+        $user = $dbForProject->getDocument('users', $userId);
+
+        if ($user->isEmpty()) {
+            throw new Exception(Exception::USER_NOT_FOUND);
+        }
+
+        $target = $dbForProject->getDocument('targets', $targetId);
+
+        if ($target->isEmpty()) {
+            throw new Exception(Exception::USER_TARGET_NOT_FOUND);
+        }
+
+        $target->setAttribute('identifier', $identifier);
```

Comment:
Let's also check that the target belongs to the user before updating

Reply:
done this